### PR TITLE
Issue #100 adview manager logic

### LINKED ISSUE #100: adex-adview-manager logic in Rust
it should be in a separate crate, but we need the ad picking logic of adex-adview-manager ported to Rust so that it can be used seamlessly on more platforms (e.g. android, iOS) 

it will reuse the Unit type from the domain crate

we're referring to `applySelection` and `getHTML`

### PR DESCRIPTION
**Pull request by elpiel**
Resolves #100

### CODE REVIEW

File: adview-manager/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+            {
+                Some(Unit::new(by_price, targeting_score))
+            } else {
+                None
+            }
+        })
+        .collect();
+    by_score.sort_by(|a, b| {
+        a.targeting_score
+            .partial_cmp(&b.targeting_score)
```
Comment (Ivshti) [inline adview-manager/src/lib.rs]: why is this returning a Result? just curious

File: adview-manager/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+            .expect("Should always be comparable")
+    });
+    by_score.truncate(options.top_by_score);
+
+    by_score
+}
+
+fn is_video(ad_unit: &AdUnit) -> bool {
+    ad_unit.media_mime.split('/').take(1).collect::<Vec<&str>>()[0] == "video"
```
Comment (Ivshti) [inline adview-manager/src/lib.rs]: you can just `.nth(1) == "video"`

File: adview-manager/src/lib.rs (hunk 1 above)
Comment (elpiel) [reply adview-manager/src/lib.rs]: Targeting score is f64, we can't directly compare them, but we can do a `PartialCmp`:
[Docs](https://doc.rust-lang.org/std/cmp/trait.PartialOrd.html#tymethod.partial_cmp)
```
let result = std::f64::NAN.partial_cmp(&1.0);
assert_eq!(result, None);
```

File: adview-manager/src/lib.rs (hunk 2 above)
Comment (elpiel) [reply adview-manager/src/lib.rs]: Done

**Comment by elpiel** [comment]
Since Clippy was actually kinda disabled and currently the validator worker is the default project + there are quite unfinished stuff that trigger clippy from the "Refactor implementation" I've forgot to run clippy on this new crate. I will do that and fix the warnings!

**Comment by elpiel** [comment]
Fixed @Ivshti

**Pull request merged**